packages/lib/CalendarService.ts

```diff
@@ -274,6 +285,24 @@ export default abstract class BaseCalendarService implements Calendar {
       if (vevent?.getFirstPropertyValue("transp") === "TRANSPARENT") return;
 
       const event = new ICAL.Event(vevent);
+
+      const tzid: string | undefined = vevent?.getFirstPropertyValue("tzid");
+      // In case of icalendar, when only tzid is available without vtimezone, we need to add vtimezone explicitly to take care of timezone diff
+      if (!vcalendar.getFirstSubcomponent("vtimezone") && tzid) {
+        const timezoneComp = new ICAL.Component("vtimezone");
+        timezoneComp.addPropertyWithValue("tzid", tzid);
+        const standard = new ICAL.Component("standard");
+        // get timezone offset
+        const tzoffsetfrom = buildUtcOffset(dayjs(event.startDate.toJSDate()).tz(tzid, true).utcOffset());
+        const tzoffsetto = buildUtcOffset(dayjs(event.endDate.toJSDate()).tz(tzid, true).utcOffset());
+        // set timezone offset
+        standard.addPropertyWithValue("tzoffsetfrom", tzoffsetfrom);
+        standard.addPropertyWithValue("tzoffsetto", tzoffsetto);
+        // provide a standard dtstart
+        standard.addPropertyWithValue("dtstart", "1601-01-01T00:00:00");
+        timezoneComp.addSubcomponent(standard);
+        vcalendar.addSubcomponent(timezoneComp);
+      }
```

Comment:
If `vTimezone` is absent and `tzid` is present, we "build" the `vTimezone` component from the `tzid` and make sure to include the `dtstart`, `tzoffsetfrom`, and `tzoffsetto` and then add it to the `vCalendar` component to ensure proper timezone conversion

Reply:
this might still require consideration of daylight savings but I believe I'd need to add that as a follow up